reverse_engineering/api.js

```diff
@@ -21,19 +21,22 @@ const { prepareError } = require('./databaseService/helpers/errorService');
 
 module.exports = {
 	async connect(connectionInfo, logger, callback, app) {
-		const client = getClient();
-		const sshService = app.require('@hackolade/ssh-service');
+		const client = clientManager.getClient();
+
 		if (!client) {
-			await setClient(connectionInfo, sshService, 0, logger);
-			return getClient();
+			await clientManager.setClient({
+				connectionInfo,
+				logger,
+				sshService: app.require('@hackolade/ssh-service'),
+			});
+			return clientManager.getClient();
 		}
 
 		return client;
 	},
 
 	disconnect(connectionInfo, logger, callback, app) {
-		const sshService = app.require('@hackolade/ssh-service');
-		clearClient(sshService);
+		clientManager.clearClient({ sshService: app.require('@hackolade/ssh-service') });
```

Comment:
Why do you need to pass the `sshService` here? You already passed it when calling `setClient()`, so the `clientManager` could just remember it...

Reply:
good idea, will preserve that